### DISCUSSION TITLE
How do you recommend saving and loading trained models?

### DISCUSSION THREAD

**Discussion by johnandersen777** [intel/scikit-learn-intelex#484] [Q&A] [answered]
For example in

https://github.com/IntelPython/daal4py/blob/a579737bcebf42b81bca5db2e0b6203682a9d98d/examples/linear_regression_streaming.py#L69

should we just pickle `train_result.model`?

**Comment by johnandersen777** [comment]
This is what [DFFML](https://github.com/intel/dffml/blob/1b0ede6dbd76772a544b5febff0c006cca939ecd/model/daal4py/dffml_model_daal4py/daal4pylr.py#L76-L77) has done:

```python
        self.lm_trained = self.lm.finalize().model
        self.joblib.dump(self.lm_trained, self.path)
```

**Comment by PivovarA** [comment, marked as answer]
Hi @pdxjohnny 
In daal each model has its own serializer. Therefore, these serializers are called during serialization. 
Therefore joblib should be enough for model saving
